desktop-widgets/configuredivecomputerdialog.cpp

```diff
@@ -290,6 +290,7 @@ void OstcFirmwareCheck::parseOstcFwVersion(QNetworkReply *reply)
 	int firstOpenBracket = parse.indexOf('[');
 	int firstCloseBracket = parse.indexOf(']');
 	latestFirmwareAvailable = parse.mid(firstOpenBracket + 1, firstCloseBracket - firstOpenBracket - 1);
+	reply->close();
```

Comment:
Is this necessary?

One would think that the reply gets destructed anyway?

Reply:
Not super sure it is strictly needed, but I think it can't hurt if we close the network connection before we move to the next step in the next step in the sequence that will do more http requests. I added this while I was trying to find why the download wasn't working - in the end it turned out to be early destruction of the `QNetworkManager` due to the closing of the dialog.

Reply:
@bstoeger: Are you ok with the way this is now, would you prefer the `close()` to be removed?

Reply:
I don't object to it - it's up to you.

Frankly, the whole resource management in this file is crazy, which is probably due to the `QNetworkAccessManager` interface being a mess. Who owns `reply`? Do you have to delete it with `deleteLater()`? I skimmed the documentation and found it frustratingly unclear. This is a horrible pre-C++11 API.

Reply:
Agreed re. mess. But also not keen to spend eons cleaning up something that will affect only a fraction of users, and will only ever be used very infrequently - this is probably why it has been broken for a long time.
Re cleaning up of `reply`, I got it on good authority that this is indeed deleted (and the connection closed) when the owning object is destroyed. This is what caused the bug in the first place, as the second connection was closed from our side before the download of the firmware file was completed.